offchain-modules/packages/x/src/ckb/tx-helper/generator.ts

```diff
@@ -802,6 +802,223 @@ export class CkbTxGenerator extends CkbTxHelper {
+
+    // add multisigChangeCell if needed
+    let omniLockMultisigChangeCellMinimalCapacity = BigInt(0);
+    if (searchedMultisigCellsAmount > totalSudtOutputAmount) {
+      const multisigChangeCellAmount = searchedMultisigCellsAmount - totalSudtOutputAmount;
+      omniLockMultisigChangeCell.data = `0x${multisigChangeCellAmount.toString(16)}`;
+      omniLockMultisigChangeCellMinimalCapacity = minimalCellCapacity(omniLockMultisigChangeCell);
+      omniLockMultisigChangeCell.cell_output.capacity = `0x${omniLockMultisigChangeCellMinimalCapacity.toString(16)}`;
+      txSkeleton = txSkeleton.update('outputs', (outputs) => {
+        return outputs.push(omniLockMultisigChangeCell);
+      });
+    }
+
+    // add collector cell to pay tx fee
+    const collectorLockscript = parseAddress(collectorAddress);
+    const collectorChangeCell: Cell = {
+      cell_output: {
+        capacity: `0x0`,
+        lock: collectorLockscript,
+      },
+      data: '0x',
+    };
+    const collectorChangeCellMinimalCapacity = minimalCellCapacity(collectorChangeCell);
+    const collectorNeedCapacity = totalSudtOutputCapacity + collectorChangeCellMinimalCapacity + 1n * BigInt(100000000);
+    const searchedCollectorCells = await this.collector.getCellsByLockscriptAndCapacity(
+      collectorLockscript,
+      collectorNeedCapacity,
+    );
+    const searchedCollectorCellsCapacity = searchedCollectorCells
+      .map((cell) => BigInt(cell.cell_output.capacity))
+      .reduce((a, b) => a + b, 0n);
+    if (searchedCollectorCellsCapacity < collectorNeedCapacity)
+      throw new Error(
+        `collector cell ckb capacity not enough, need ${collectorNeedCapacity.toString()}, found ${searchedCollectorCellsCapacity.toString()}`,
+      );
+    txSkeleton = txSkeleton.update('inputs', (inputs) => {
+      return inputs.push(...searchedCollectorCells);
+    });
+    collectorChangeCell.cell_output.capacity = `0x${(
+      searchedMultisigCellsCapacity +
+      searchedCollectorCellsCapacity -
+      totalSudtOutputCapacity -
+      omniLockMultisigChangeCellMinimalCapacity
+    ).toString(16)}`;
+    txSkeleton = txSkeleton.update('outputs', (outputs) => {
+      return outputs.push(collectorChangeCell);
+    });
+
+    // add witness
+    const omniLockWitnessPlaceholder = getOmniLockMultisigWitnessPlaceholder();
+    const omniLockWitnessGroup = lodash.range(searchedMultisigCells.length).map((index) => {
+      if (index === 0) return omniLockWitnessPlaceholder;
+      return '0x';
+    });
+    txSkeleton = txSkeleton.update('witnesses', (witnesses) => {
+      return witnesses.push(...omniLockWitnessGroup);
+    });
+    const collectorWitness = new Reader(
+      SerializeWitnessArgs(
+        normalizers.NormalizeWitnessArgs({
+          lock: SECP_SIGNATURE_PLACEHOLDER,
+        }),
+      ),
+    ).serializeJson();
+    txSkeleton = txSkeleton.update('witnesses', (witnesses) => {
+      return witnesses.push(collectorWitness);
+    });
```

Comment:
please fix this refer to https://github.com/nervosnetwork/force-bridge/pull/285

Reply:
fixed